core/src/main/java/com/datastax/oss/driver/internal/core/graal/DefaultFeature.java

```diff
@@ -0,0 +1,122 @@
+ * instances to be instantiated via reflection, adding each to the current reflection config.
+ */
+@AutomaticFeature
+final class DefaultFeature implements Feature {
+
+  public Config buildConfig() {
+
+    /* Copied from DefaultDriverConfigLoader, including the DEFAULT_ROOT_PATH value there as the literal
+    "datastax-java-driver" below.  Can't reference the class directly here since that will introduce
+    a requirement to specify it as a build-time dependency (as well as the slf4j logger it creates
+    via static init */
+    return ConfigFactory.defaultOverrides()
+        .withFallback(ConfigFactory.defaultApplication())
+        .withFallback(ConfigFactory.defaultReference(CqlSession.class.getClassLoader()))
+        .resolve()
+        .getConfig("datastax-java-driver");
+  }
+
+  public Optional<String> tryString(Config config, DefaultDriverOption option) {
+
+    try {
+      return Optional.of(config.getString(option.getPath()));
+    } catch (ConfigException.WrongType e) {
+      return Optional.empty();
+    } catch (ConfigException.Missing e) {
+      return Optional.empty();
+    }
+  }
+
+  public Optional<Class<?>> tryLoadClass(String configVal, String... defaultPackages) {
+
+    try {
+      if (configVal.contains(".")) {
+        return Optional.of(Class.forName(configVal));
+      } else {
+        for (String defaultPackage : defaultPackages) {
+          String qualifiedClassName = defaultPackage + "." + configVal;
+          try {
+            return Optional.of(Class.forName(qualifiedClassName));
+          } catch (ClassNotFoundException cnfe) {
+          }
+        }
+        return Optional.empty();
+      }
+    } catch (LinkageError | Exception e) {
+      return Optional.empty();
+    }
+  }
+
+  public void tryRegister(Config config, DefaultDriverOption option, String... defaultPackages) {
+
+    tryString(config, option)
+        .flatMap(configVal -> tryLoadClass(configVal, defaultPackages))
+        .ifPresent(
+            (clz) -> {
+              /* Mark the class as being initialized at build time */
+              RuntimeClassInitialization.initializeAtBuildTime(clz);
+
+              /* Register all the things as being available at runtime */
+              RuntimeReflection.register(clz);
+              for (Method m : clz.getMethods()) {
+                RuntimeReflection.register(m);
+              }
+              for (Field f : clz.getFields()) {
+                RuntimeReflection.register(f);
+              }
+              for (Constructor c : clz.getConstructors()) {
+                RuntimeReflection.register(c);
+              }
+            });
+  }
+
+  @Override
+  public void beforeAnalysis(Feature.BeforeAnalysisAccess access) {
+
+    /* Make the Typesafe classes we need to do our work available at build-time */
+    RuntimeClassInitialization.initializeAtBuildTime("com.typesafe.config.impl");
```

Comment:
We use the Typesafe classes to read the config above and since this feature executes at build time those classes must be available for build-time initialization.

This is just a placeholder for now until we implement a more generic way for dealing with this problem.

Reply:
Gotcha. In fact, this class should _probably_ start by detecting whether TyepSafe is available or not. If it is, then it should go on and load the config; if it is not, it could do two things IMO:

1. Nothing, and return immediately.
2. Resort to a best-effort thing and register for reflection a predefined set of built-in impls (the ones being discussed [here](https://github.com/datastax/java-driver/pull/1560/files#r665605797)).

Wdyt?